Core/include/Acts/Utilities/Any.hpp

```diff
@@ -0,0 +1,495 @@
+    }
+    return *this;
+  }
+
+  AnyBase(AnyBase&& other) {
+    _ACTS_ANY_VERBOSE(
+        "Move construct (this=" << this << ") at: " << (void*)m_data.data());
+    if (m_handler == nullptr && other.m_handler == nullptr) {
+      // both are empty, noop
+      return;
+    }
+
+    m_handler = other.m_handler;
+    assert(m_handler != nullptr && "Handler is null");
+    moveConstruct(std::move(other));
+  }
+
+  AnyBase& operator=(AnyBase&& other) {
+    _ACTS_ANY_VERBOSE("Move assign (this=" << this
+                                           << ") at: " << (void*)m_data.data());
+    if (m_handler == nullptr && other.m_handler == nullptr) {
+      // both are empty, noop
+      return *this;
+    }
+
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler && "Handler is null");
+      moveConstruct(std::move(other));
+    } else {
+      // @TODO: Support assigning between different types
+      if (m_handler != other.m_handler) {
+        throw std::bad_any_cast{};
+      }
+      move(std::move(other));
+    }
+    return *this;
+  }
+
+  operator bool() const { return m_handler != nullptr; }
+
+ private:
+  void* dataPtr() {
+    if (m_handler->heap) {
+      return *reinterpret_cast<void**>(m_data.data());
+    } else {
+      return reinterpret_cast<void*>(m_data.data());
+    }
+  }
+
+  void setDataPtr(void* ptr) { *reinterpret_cast<void**>(m_data.data()) = ptr; }
+
+  const void* dataPtr() const {
+    if (m_handler->heap) {
+      return *reinterpret_cast<void* const*>(m_data.data());
+    } else {
+      return reinterpret_cast<const void*>(m_data.data());
+    }
+  }
+
+  struct Handler {
+    void (*destroy)(void* ptr) = nullptr;
+    void (*moveConstruct)(void* from, void* to) = nullptr;
+    void (*move)(void* from, void* to) = nullptr;
+    void* (*copyConstruct)(const void* from, void* to) = nullptr;
+    void (*copy)(const void* from, void* to) = nullptr;
+    bool heap{false};
+  };
+
+  template <typename T>
+  static Handler* makeHandler() {
+    static_assert(!std::is_same_v<T, AnyBase<SIZE>>, "Cannot wrap any in any");
+    static Handler static_handler = []() {
+      Handler h;
+      if constexpr (!std::is_trivially_destructible_v<T> || sizeof(T) > SIZE) {
+        h.destroy = &destroyImpl<T>;
+      }
+      if constexpr (!std::is_trivially_move_constructible_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.moveConstruct = &moveConstructImpl<T>;
+      }
+      if constexpr (!std::is_trivially_move_assignable_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.move = &moveImpl<T>;
+      }
+      if constexpr (!std::is_trivially_copy_constructible_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.copyConstruct = &copyConstructImpl<T>;
+      }
+      if constexpr (!std::is_trivially_copy_assignable_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.copy = &copyImpl<T>;
+      }
+      h.heap = sizeof(T) > SIZE;
```

Comment:
Consider setting `h.heap` in the beginning and replacing the repetitive `sizeof(T) > SIZE` conditional with `h.heap`. Also, maybe `heapAllocated` would be a clearer name, but naming is subjective, so up to you.

Reply:
Also, you may be able to finish removing this recurring conditional from the rest of the code by adding one tiny utility function to replace it everywhere:

```c++
template <typename T>
static bool heapAllocated() {
    return sizeof(T) > SIZE;
}
```

Reply:
To the wirst point, I can't use `h.heapAllocated` in the ifs because it's not constexpr.

Reply:
You can work around that limitation by using an intermediary constexpr value:

```c++
static constexpr bool heapAllocated = sizeof(T) > SIZE;
h.heapAllocated = heapAllocated;
if constexpr (!std::is_trivially_destructible_v<T> || heapAllocated) {
    /* ...and so on ... */
```

Reply:
Then again, the current code that runs heapAllocated<T>() multiple times isn't that bad.